lib/migration-payloads/validation/schema-validation.js

```diff
@@ -132,6 +132,38 @@ const validateContentType = function (payload) {
   });
 };
 
+const unknownCombinationError = function ({ path, keys }) {
+  const type = 'object.unknownCombination';
+  const message = `"${keys.join()}" is not allowed`;
+  const context = { keys };
+
+  return { message, path, type, context };
+};
+
+const combineErrors = function (fieldValidationsErrors) {
```

Comment:
Maybe you could add a comment that shows the structure we're dealing with for every step of transformations so it's more easily understandable.